### DISCUSSION TITLE
[Solved] Error out-of-memory after upgrade to 1.25.0 on RasPi 1B (for info)

### DISCUSSION THREAD

**Discussion by Isy99** [Koenkk/zigbee2mqtt#12214] [General]
Hello folks, 
my first post here - great work and many thanks for the tool!

I updated my  installation on an old PI1B (yes, I'm aware of the limitations) from 1.24.0 to 1.25.0 and after `npm start` I receive an out-of-memory message.

```
pi@zigbee:/opt/zigbee2mqtt $ npm start
> zigbee2mqtt@1.25.0 start
> node index.js
Building Zigbee2MQTT... (initial build), failed
Build failed; ran out-of-memory, free some memory (RAM) and start again
Error: Command failed: npm run build
<--- Last few GCs --->
[546:0x4247290]   480594 ms: Mark-sweep 123.5 (129.3) -> 123.1 (129.3) MB, 4500.0 / 0.1 ms  (average mu = 0.563, current mu = 0.113) allocation failure scaven                 ge might not succeed
[546:0x4247290]   487189 ms: Mark-sweep 123.6 (129.3) -> 123.3 (129.6) MB, 5315.2 / 0.1 ms  (average mu = 0.421, current mu = 0.194) allocation failure scaven                 ge might not succeed
<--- JS stacktrace --->
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
Aborted
```

I tried different options mentioned in RasPi forums, nothing helped. 
I downgraded to 1.24.0 and succeded starting zigbee2mqtt.

Kind regards,
Helmut

**Comment by Impact123** [comment]
Run `export NODE_OPTIONS=--max_old_space_size=256` before executing `npm start`.
See #12034.
If that doesn't work you can try with `128`, stop other processes, try increasing swap and/or install zram.

**Reply by Isy99** [reply]
Wow - that was a fast replay! Mni thanks! And sorry, I did not find the thread you mentioned. Next time I try longer -:)
I'll give it a try.

**Reply by Isy99** [reply]
Back online with 1.25.0.
Great support, thanks.
